internal/error/errors.go

```diff
@@ -0,0 +1,84 @@
+package error
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	installConflictErrorPattern        = `Unable to continue with install: (\w+) "(.*)" in namespace "(.*)" exists and cannot be imported into the current release.*`
+	installConflictErrorGeneralPattern = `Unable to continue with install: .*`
+)
+
+// InstallConflictErr indicates one or more resources already exist in the cluster
+type InstallConflictErr error
+
+// UnknownErr indicates an error that cannot be classified
+type UnknownErr error
+
+func AsOlmErr(err error) error {
+	if err == nil {
+		return nil
+	}
+	for _, rule := range rules {
+		if matches := rule.MatchError(err); len(matches) > 0 {
+			// first index is the full match
+			logrus.Debugf("found match with params %v\n", matches)
+			return rule.TranslateError(err, matches[1:]...)
+		}
+	}
+
+	// let's mark any unmatched errors as unknown
+	return UnknownErr(err)
```

Comment:
`UnknownErr` is not explicitly checked anywhere as a type. Do we need this. Could this simply `return err`?